packages/s2-core/src/utils/export/copy.ts

```diff
@@ -111,17 +111,66 @@ const getHeaderList = (headerId: string) => {
   return headerList;
 };
 
-// 把 string[][] 矩阵转换成字符串格式
-const transformDataMatrixToStr = (dataMatrix: string[][]) => {
-  return map(dataMatrix, (line) => line.join(newTab)).join(newLine);
+type MatrixTransformer = (data: string[][]) => CopyableItem;
+
+type CopyMIMEType = 'text/plain' | 'text/html';
+
+export type CopyableItem = {
+  type: CopyMIMEType;
+  content: string;
+};
+
+export type Copyable = CopyableItem | CopyableItem[];
+
+function pickDataFromCopyable(
+  copyable: Copyable,
+  type: CopyMIMEType | CopyMIMEType[] = 'text/plain',
+): string | string[] {
+  if (Array.isArray(type)) {
+    return ([].concat(copyable) as CopyableItem[])
+      .filter((item) => type.includes(item.type))
+      .map((item) => item.content);
+  }
+
+  return (
+    ([].concat(copyable) as CopyableItem[])
+      .filter((item) => item.type === type)
+      .map((item) => item.content)[0] || ''
+  );
+}
+
+// 把 string[][] 矩阵转换成 CopyableItem
+const matrixPlainTextTransformer: MatrixTransformer = (dataMatrix) => {
+  return {
+    type: 'text/plain',
+    content: map(dataMatrix, (line) => line.join(newTab)).join(newLine),
+  };
+};
+
+// 把 string[][] 矩阵转换成 CopyableItem
+const matrixHtmlTransformer: MatrixTransformer = (dataMatrix) => {
+  function getCells(data: string[], tagName) {
```

Comment:
感觉这个函数更像创建一行
```suggestion
  function createTableRow(data: string[], tagName) 
```
可以将 `tr` 的拼接也放过来。